src/jdk.jlink/share/classes/jdk/tools/jlink/internal/TaskHelper.java

```diff
@@ -566,7 +565,7 @@ public boolean shouldListPlugins() {
                         if (option.isTerminal()) {
                             return ++i < args.length
                                         ? Stream.of(Arrays.copyOfRange(args, i, args.length))
-                                                .collect(Collectors.toList())
+                                                .toList()
                                         : Collections.emptyList();
```

Comment:
This is probably out of scope for a simple toList() conversion, but there are some additional ways this code could be simplified. It's possible to stream a subrange of an array directly:

`Arrays.stream(args, i, args.length).toList()`

Since a zero-length array subrange results in a valid zero-length list, the enclosing ternary is unnecessary:

`return Arrays.stream(args, ++i, args.length).toList();`

It's also possible to do this without using streams at all,

`return List.copyOf(Arrays.asList(args).subList(++i, args.length));`

but this actually seems more cumbersome than streaming the array subrange. In any case, perhaps this sort of cleanup could be reserved for a future changeset from the jlink maintainer.

Reply:
Agree that it's less cumbersome, but will keep it for a future clean up.